main/job-counter.js

```diff
@@ -0,0 +1,56 @@
+'use strict'
+
+const Store = require('electron-store')
+
+const jobCounterStore = new Store({
+  name: 'job-counters'
+})
+
+class JobCounter {
+  #moduleCounters
+
+  constructor () {
+    this.#moduleCounters = loadStoredCounters()
+  }
+
+  getTotalJobCount () {
+    return Object.values(this.#moduleCounters).reduce((sum, value) => sum + value, 0)
+  }
+
+  /**
+   * @param {string} moduleName
+   * @param {number} count
+   */
+  setModuleJobCount (moduleName, count) {
+    this.#moduleCounters[moduleName] = count
+    this.#save()
+  }
+
+  reset () {
+    this.#moduleCounters = {}
+    this.#save()
+  }
+
+  #save () {
+    storeCounters(this.#moduleCounters)
+  }
+}
+
+/**
+ * @returns {import('./typings').ModuleJobCount}
+ */
+function loadStoredCounters () {
+  // A workaround to fix false TypeScript errors
+  return /** @type {any} */(jobCounterStore.get('counters', {}))
+}
+
+/**
+ * @param {import('./typings').ModuleJobCount} counters
```

Comment:
could we move the type imports to the file header, for readability?

Reply:
Of course.

I am usually adding these type names via VS Code IntelliSense, that's why we are getting these inline imports. 🤷🏻